File: simple_li_re.py
#Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 

#Importing the datasets
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:,:-1].values
Y = dataset.iloc[:,1].values



#Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.2,random_state=0)

#Training Simple Linear Regression
from sklearn.linear_model import LinearRegression
regressor =LinearRegression()
regressor.fit(X_train,Y_train)

#predicting the test set results
y_pred=regressor.predict(X_test)

#visualising training set
plt.scatter(X_train,Y_train,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
plt.title('Salary vs Experience(Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

#visualising test set
plt.scatter(X_test,Y_test,color='red')
plt.plot(X_train,regressor.predict(X_train),color='blue')
#plt.scatter(X_test,y_pred,color='green')
plt.title('Salary vs Experience(Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: simple_li_re.R
#Simple Linear Regression

#Importing the dataset
dataset = read.csv('Salary_Data.csv')
#dataset=dataset[,2:3]


#Splitting the dataset into Training set and Testing set
#install.packages("caTools")
library(caTools)
set.seed(123)
split=sample.split(dataset$Salary,SplitRatio = 2/3)
training_set=subset(dataset,split==TRUE)
test_set=subset(dataset,split==FALSE)
 
#Fitting Simple Linear Regression to training set
regressor = lm(formula = Salary ~ YearsExperience,data = training_set)

#predicting the Test set results
y_pred=predict(regressor,newdata = test_set)

#Visualising the training set results
# install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary),colour='red')+
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,newdata = training_set)),colour='blue')+
  ggtitle('Years of Experience vs Salary(Training Set)')+
  xlab('Years of Experience')+
  ylab('Salary')

#Visualising the test set results
# install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=test_set$YearsExperience,y=test_set$Salary),colour='red')+
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,newdata = training_set)),colour='blue')+
  ggtitle('Years of Experience vs Salary(Test Set)')+
  xlab('Years of Experience')+
  ylab('Salary')
